mlflow/pipelines/utils/execution.py

```diff
@@ -400,6 +400,19 @@ def _create_makefile(pipeline_root_path, execution_directory_path, template) ->
 
     if template == "regression/v1":
         makefile_to_use = _REGRESSION_MAKEFILE_FORMAT_STRING
+        for required_file in [
+            "ingest.py",
+            "split.py",
+            "train.py",
+            "transform.py",
+            "custom_metrics.py",
+        ]:
```

Comment:
I am not sure if all of these files are required if the pipeline config doesn't have them mentioned in the yaml config. These could be optional. But if they are specified, I agree we should have them. So can check for that here? 
Let me know if I am missing something. 
cc @mengxr

Reply:
They are required by the current implementation even if unspecified, though I agree it would be desirable for them only be required if specified. Perhaps we could make that change separately, at any rate I think this PR is an improvement on what exists today. Nice work @mingyu89 !